src/components/Canvas/Editor.tsx

```diff
@@ -569,6 +586,56 @@ export default class Editor extends EventDispatcher {
     this.interactionLayer.setDpr(dpr);
   }
 
+  /**
+   * @summary Sets the data of the pixel array (use with caution, since data will be overwritten!)
+   * @param data Array of PixelModifyItem (It must be a rectangular array, i.e. all rows must have the same length)
+   */
+  setData(data: Array<Array<PixelModifyItem>>) {
+    const { isDataValid, rowCount, columnCount } = validateSquareArray(data);
+    if (!isDataValid) {
+      throw new Error(`Data is not valid`);
+    }
+    const leftColumnIndex = data[0][0].columnIndex;
+    const topRowIndex = data[0][0].rowIndex;
+
+    // reset data
+    this.dataLayer.setData(new Map());
+    for (let i = 0; i < data.length; i++) {
+      this.dataLayer.getData().set(topRowIndex + i, new Map());
+      for (let j = 0; j < data[i].length; j++) {
+        this.dataLayer
+          .getData()
+          .get(topRowIndex + i)!
```

Comment:
question) why is non null assertion used here?

Reply:
It seems unnecessary. I will delete the assertion!